stock_quant_package_restrict_disposable_reuse/models/stock_move_line.py

```diff
@@ -0,0 +1,80 @@
+# 2024 Copyright ForgeFlow, S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    result_package_id = fields.Many2one(
+        domain="['|', '|', ('location_id', '=', False), ('location_id', '=', location_dest_id),"
+        " ('id', '=', package_id), '|', ('package_type_id.restrict_reuse', '=', False),"
+        " ('disposable_used', '=', False)]"
+    )
+    package_id = fields.Many2one(
+        domain="[('location_id', '=', location_id), '|', "
+        "('package_type_id.restrict_reuse', '=', False), ('disposable_used', '=', False)]"
+    )
+
+    @api.constrains("package_id", "result_package_id")
+    def _check_disposable_package_constrain(self):
+        for record in self:
+            record._check_disposable_packages()
+
+    def _check_disposable_packages(self):
+        if (
+            self.package_id
+            and self.package_id.package_type_id.restrict_reuse
+            and self.package_id.package_use == "disposable"
+            and self.package_id.disposable_used
+        ):
+            raise ValidationError(
+                _(
+                    "You can't reuse package %s because it is a disposable package that has "
```

Comment:
```suggestion
                    "You can't reuse package %(package_name)s because it is a disposable package that has "
                    "already been used in a delivery.", pacakge_name=self.package_id.name
```